libraries/gpu-gl-common/src/gpu/gl/GLBackend.cpp

```diff
@@ -840,10 +840,7 @@ void GLBackend::do_glColor4f(const Batch& batch, size_t paramOffset) {
     if (_input._colorAttribute != newColor) {
         _input._colorAttribute = newColor;
         glVertexAttrib4fv(gpu::Stream::COLOR, &_input._colorAttribute.r);
-        // Color has been changed and is not white. To prevent colors from bleeding
-        // between different objects, we need to set the _hadColorAttribute flag
-        // as if a previous render call had potential colors
-        _input._hadColorAttribute = (newColor != glm::vec4(1.0f, 1.0f, 1.0f, 1.0f));
+        _input._hasColorAttribute = true;
```

Comment:
`renderShape` calls the old hacky batch._glColor4f, which revealed it had a bug whereby the first object would render white.  I tried fixing it up